compiler/rustc_hir/src/hir.rs

```diff
@@ -945,7 +945,21 @@ pub struct Closure<'hir> {
     pub fn_decl_span: Span,
     /// The span of the argument block `|...|`
     pub fn_arg_span: Option<Span>,
-    pub movability: Option<Movability>,
+    pub kind: ClosureKind,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Copy, Hash, HashStable_Generic, Encodable, Decodable)]
+pub enum ClosureKind {
+    /// This is a plain closure expression.
+    Closure,
+    /// This is a coroutine expression -- i.e. a closure expression in which
+    /// we've found a `yield`. These can arise either from "plain" coroutine
+    ///  usage (e.g. `let x = || { yield (); }`) or from a desugared expression
+    /// (e.g. `async` and `gen` blocks).
+    // FIXME(coroutines): We could probably remove movability here -- it can be deduced
+    // from the `CoroutineKind` in all cases (except for "plain" coroutines, which could
+    // carry the movability in the variant).
```

Comment:
This `FIXME` has been taken care of already, right?

Reply:
Yes, oop